plugins/mergeStyles.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+exports.type = 'full';
+exports.active = true;
+exports.description = 'merge multiple style elements into one';
+
+var cssTools = require('../lib/css-tools');
+
+/**
+ * Merge multiple style elements into one.
+ *
+ * @param {Object} document document element
+ *
+ * @author strarsis <strarsis@gmail.com>
+ */
+exports.fn = function (document) {
+  // collect <style/>s (preserve order)
+  var styleEls = document.querySelectorAll('style');
+
+  // no <styles/>s, nothing to do
+  if (styleEls === null || styleEls.length <= 1) {
+    return document;
+  }
+
+  var styles = [];
+  for (var styleEl of styleEls) {
+    if (styleEl.isEmpty() || styleEl.closestElem('foreignObject')) {
+      // skip empty <style/>s or <foreignObject> content.
+      continue;
+    }
+
+    var cssStr = cssTools.getCssStr(styleEl);
```

Comment:
Use const where possible or let if you need mutation

Reply:
OK, should be const where it isn't mutated.